### ISSUE TITLE
Map-Controller: _viewportProps propagation to makeViewport()

### ISSUE THREAD

**Issue by sumarlidason** [visgl/deck.gl#5430]
#### Description
_This might not be the most well formed bug report, but hopefully there is enough information here._

`makeViewport` functions as intended when executed from `view-manager`, but when executed from `map-controller` during pan/zoom results in a missing or `NaN` filled matrix.

![image](https://user-images.githubusercontent.com/1991770/106782820-9cbfdd00-6618-11eb-9898-a9b09fcf6f5e.png)

In the latter case when `_getViewport ` is called `viewportDimensions` is `undefined` causing raw css props for `width` and `height` to passed down the stack instead of pixel dimensions:

https://github.com/visgl/deck.gl/blob/44e709651ab8a727c8d04a3df8faba75eb6c0e50/modules/core/src/views/view.js#L150-L154

* you can see here `viewportDimensions` are supplied and render succeeds `840` & `375`:
<img width="1405" alt="defd" src="https://user-images.githubusercontent.com/1991770/106785386-53bd5800-661b-11eb-8fe6-6aaa2745b853.png">

* here, the `viewportDimensions` are missing and the raw props are passed down `50%` & `100%`:
<img width="1376" alt="Screen Shot 2021-02-03 at 12 27 01 PM" src="https://user-images.githubusercontent.com/1991770/106785324-4738ff80-661b-11eb-9bc1-792941d8ac70.png">

#### JSX
```js
<DeckGL viewState={viewState} controller={false}>
  <MapView id={'mapview-left'} x={0} y={0} width="50%" height="100%" controller={true} 
            onViewStateChange={this.updateViewport}>
    <StaticMap
      visible={true}
      key={'streets'}
      mapStyle={C[`MAPBOX_STYLE_${theme.palette.type.toUpperCase()}_BASE`]}
      mapboxApiAccessToken={mapboxAccessToken}
    />
  </MapView>
</DeckGL>
```

#### Expected Behavior
Functioning map load, pan, zoom etc.

#### Repro Steps

TODO: CodePen

#### Environment
<details>
<summary>System</summary>

```
  System:
    OS: macOS 11.1
  Binaries:
    Node: 14.15.4 - ~/.nvm/versions/node/v14.15.4/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.11 - ~/.nvm/versions/node/v14.15.4/bin/npm
  Browsers:
    Chrome: 88.0.4324.96
    Edge: Not Found
    Firefox: Not Found
    Safari: 14.0.2
```

</details>

<details>
<summary>Modules</summary>

```
    "@deck.gl/aggregation-layers": "^8.4.0",
    "@deck.gl/core": "^8.4.0",
    "@deck.gl/extensions": "^8.4.0",
    "@deck.gl/geo-layers": "^8.4.0",
    "@deck.gl/google-maps": "^8.4.0",
    "@deck.gl/json": "^8.4.0",
    "@deck.gl/layers": "^8.4.0",
    "@deck.gl/mapbox": "^8.4.0",
    "@deck.gl/mesh-layers": "^8.4.0",
    "@deck.gl/react": "^8.4.0",
    "@luma.gl/core": "^8.4.0",
    "@luma.gl/engine": "^8.4.0",
    "@luma.gl/gltools": "^8.4.0",
    "@luma.gl/webgl": "^8.4.0",
    
    "react-map-gl": "^6.1.2",
    "react": "^16.13.1",
```

</details>

#### Hack
```diff
diff --git a/modules/core/src/controllers/map-controller.js b/modules/core/src/controllers/map-controller.js
index 9af1a7b6a..64c3ec997 100644
--- a/modules/core/src/controllers/map-controller.js
+++ b/modules/core/src/controllers/map-controller.js
@@ -231,7 +231,7 @@ export class MapState extends ViewState {

     const zoom = this._calculateNewZoom({scale, startZoom});

-    const zoomedViewport = this.makeViewport({...this._viewportProps, zoom});
+    const zoomedViewport = this._makeViewport({...this._viewportProps, zoom});
     const [longitude, latitude] = zoomedViewport.getMapCenterByLngLatPosition({
       lngLat: startZoomLngLat,
       pos
@@ -339,7 +339,7 @@ export class MapState extends ViewState {
   _getUpdatedState(newProps) {
     // Update _viewportProps
     return new this.constructor({
-      makeViewport: this.makeViewport,
+      makeViewport: this._makeViewport,
       ...this._viewportProps,
       ...this._state,
       ...newProps
@@ -355,20 +355,29 @@ export class MapState extends ViewState {
     // Ensure pitch is within specified range
     const {maxPitch, minPitch, pitch} = props;
     props.pitch = clamp(pitch, minPitch, maxPitch);
-
     Object.assign(props, normalizeViewportProps(props));

     return props;
   }

+  _makeViewport() {
+    // this._viewportProps contains width & height as CSS percentiles, provide overrides for pixel units w/ second parameter:
+    // ref https://github.com/visgl/deck.gl/blob/c32367d8c4f808458abc90d5b2afaad43c49ab0a/modules/core/src/views/view.js#L150
+    //
+    return this.makeViewport(this._viewportProps, {
+      height: this._viewportProps.height,
+      width: this._viewportProps.width
+    });
+  }
+
   _unproject(pos) {
-    const viewport = this.makeViewport(this._viewportProps);
+    const viewport = this._makeViewport(this._viewportProps);
     return pos && viewport.unproject(pos);
   }

   // Calculate a new lnglat based on pixel dragging position
   _calculateNewLngLat({startPanLngLat, pos}) {
-    const viewport = this.makeViewport(this._viewportProps);
+    const viewport = this._makeViewport(this._viewportProps);
     return viewport.getMapCenterByLngLatPosition({lngLat: startPanLngLat, pos});
   }

@@ -442,7 +451,11 @@ export default class MapController extends Controller {
           ...LINEAR_TRANSITION_PROPS,
           transitionInterpolator: new LinearInterpolator({
             ...opts,
-            makeViewport: this.controllerState.makeViewport
+            makeViewport: props =>
+              this.controllerState.makeViewport(props, {
+                width: this.controllerStateProps.width,
+                height: this.controllerStateProps.height
+              })
           })
         }
       : LINEAR_TRANSITION_PROPS;
```

**Comment by Pessimistress** [comment]
Thanks for reporting! This should be fixed in 8.4.1.

**Comment by sumarlidason** [comment]
Thanks for the help & clean fix  @Pessimistress
